sentry_sdk/integrations/openai.py

```diff
@@ -0,0 +1,228 @@
+    completion_tokens = 0
+    prompt_tokens = 0
+    total_tokens = 0
+    if hasattr(response, "usage"):
+        if hasattr(response.usage, "completion_tokens") and isinstance(
+            response.usage.completion_tokens, int
+        ):
+            completion_tokens = response.usage.completion_tokens
+        if hasattr(response.usage, "prompt_tokens") and isinstance(
+            response.usage.prompt_tokens, int
+        ):
+            prompt_tokens = response.usage.prompt_tokens
+        if hasattr(response.usage, "total_tokens") and isinstance(
+            response.usage.total_tokens, int
+        ):
+            total_tokens = response.usage.total_tokens
+
+    if prompt_tokens == 0:
+        for message in messages:
+            if hasattr(message, "content"):
+                prompt_tokens += count_tokens(message.content)
+            elif "content" in message:
+                prompt_tokens += count_tokens(message["content"])
+
+    if completion_tokens == 0:
+        if streaming_message_responses is not None:
+            for message in streaming_message_responses:
+                completion_tokens += count_tokens(message)
+        elif hasattr(response, "choices"):
+            for choice in response.choices:
+                if hasattr(choice, "message"):
+                    completion_tokens += count_tokens(choice.message)
+
+    if total_tokens == 0:
+        total_tokens = prompt_tokens + completion_tokens
+
+    if completion_tokens != 0:
+        span.set_data(COMPLETION_TOKENS, completion_tokens)
+    if prompt_tokens != 0:
+        span.set_data(PROMPT_TOKENS, prompt_tokens)
+    if total_tokens != 0:
+        span.set_data(TOTAL_TOKENS, total_tokens)
+
+
+def _wrap_chat_completion_create(f):
+    # type: (Callable[..., Any]) -> Callable[..., Any]
+    @wraps(f)
+    def new_chat_completion(*args, **kwargs):
+        # type: (*Any, **Any) -> Any
+        hub = Hub.current
+        integration = hub.get_integration(OpenAIIntegration)
+        if integration is None:
+            return f(*args, **kwargs)
+
+        if "messages" not in kwargs:
+            # invalid call (in all versions of openai), let it return error
+            return f(*args, **kwargs)
+
+        try:
+            iter(kwargs["messages"])
+        except TypeError:
+            # invalid call (in all versions), messages must be iterable
+            return f(*args, **kwargs)
+
+        kwargs["messages"] = list(kwargs["messages"])
+        messages = kwargs["messages"]
+        model = kwargs.get("model")
+        streaming = kwargs.get("stream")
+
+        span = hub.start_span(op="openai", description="Chat Completion")
+        span.__enter__()
+        res = f(*args, **kwargs)
+        with capture_internal_exceptions():
+            span.set_data("messages", messages)
+            span.set_tag("model", model)
+            span.set_tag("streaming", streaming)
```

Comment:
SDKs should not set tags anymore. (because they are expensive to index). Please use `set_data` instead. 
If it is necessary for the product to have tags for this, we have to add something to `relay` to create the tags from the data. 

Please also check here if your data could match existing span data fields: https://develop.sentry.dev/sdk/performance/span-data-conventions/